### DISCUSSION TITLE
Any way to detect when reauthentication is needed?

### DISCUSSION THREAD

**Discussion by ShawnCBerg** [tsightler/ring-mqtt#763] [Support Q&A]
I have some automations that change the volume of a Ring Base Station, and this is only possible when the auth token is valid from what I've read. Is there a way to detect when reauth is necessary so I can send myself a notification to reauth manually to prevent these automations from not working unexpectedly?

**Comment by tsightler** [comment]
In normal operation, it should pretty much never be required to manually re-authenticate.  Tokens are long-lived, and refreshed every hour, using the same method used by the Ring app.  I've found that about 99% of cases where people think they need to re-authenticate they really just need to restart ring-mqtt.  Sometimes, for example, if Ring has an outage, the auth will get kicked out and the app will not be able to recover, even though it has code that attempts to do so.  However, a re-start will almost always fix this, I'm not sure why people immediately jump to re-authenticating or restarting HA or other things and don't just restart ring-mqtt itself.

I quite literally have only re-authenticated my production setup once in 3 years, and that was because I manually removed the auth from Ring Control Center.

This closest you can get to what you want is monitoring if the device is offline or not as any time the websocket is disconnected all devices are set as offline, however, this could happen for many reasons, not just for authentication.

**Reply by ShawnCBerg** [reply]
Thanks for the insight. The base station volume entity had been unavailable for about a week now, even through a few HA restarts as well as a full reboot. The only way I got it back online was forcing reauthentication. It sounds like this shouldn't be an issue ongoing, so I won't worry too much about it. This is the second instance that entity has gone unavailable in the last month or so, but last time it eventually fixed itself (likely after a reboot/restart).

**Reply by tsightler** [reply]
Restarting HA is almost never the answer because restarting HA does not restart addons (something it appears 99% of people don't know).  Usually you can just restart the addon, the button is right there in the UI, and the logs show everything, there's nothing hidden as to what is happening.

**Reply by ShawnCBerg** [reply]
I did not know that! I wasn't restarting HA or rebooting for this issue, but I just noticed the base station volume entity stayed offline throughout the restarts/reboots. I'll keep an eye out and just restart the add-on in the future, if necessary.
